### DISCUSSION TITLE
World Angle to Common Space angle

### DISCUSSION THREAD

**Discussion by euzu** [visgl/deck.gl#6465] [Q&A] [answered]
I am trying to draw figures with world coordinates in deck.gl with a rotation. The rotation takes place in the vertex shader. The further you go to the poles a deviation from the real angle becomes visible. I think I would have to convert the world angle to the common space angle first and then do the rotation? Is this correct? and is there a helper function I can use to do this?

**Comment by ibgreen** [comment, marked as answer]
Something similar is done in nebula.gl's PathMarkerLayer via `projectFlat`: 

https://github.com/uber/nebula.gl/blob/81c8372acf2032d27e43db802f65b3d8a6cf86b6/modules/layers/src/layers/path-marker-layer/create-path-markers.ts#L71

**Reply by euzu** [reply]
Thanks for your quick reply, I will try to understand and apply it to my project.

**Comment by euzu** [comment]
when i am rigth,  i have to calculate a point p' with the angle a in a custom distance, then use the points p and p' to calculate the angle a' in the  common space with flatten projection of p and p'.
